.github/workflows/publish-branch.yaml

```
@@ -0,0 +1,77 @@
+name: Publish Branch
+
+on:
+    push:
+        branches:
+            - main
+            - dev
+
+jobs:
+    release_candidate:
+        runs-on: ubuntu-latest
+        env:
+            NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+            NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.head_ref }}
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: "16.x"
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "yarn"
```

We use npm on the project, not yarn.
```suggestion
                  cache: "npm"
```